doc/api/esm.md

```diff
@@ -239,11 +241,30 @@ const { default: barData } =
   await import('./bar.json', { assert: { type: 'json' } });
 ```
 
-Node.js supports the following `type` values:
+Node.js supports the following `type` values, for which the assertion is
+mandatory:
+
+| Assertion `type` | Needed for       |
+| ---------------- | ---------------- |
+| `'json'`         | [JSON modules][] |
+| `'webassembly'`  | [Wasm modules][] |
 
-| `type`   | Resolves to      |
-| -------- | ---------------- |
-| `'json'` | [JSON modules][] |
+Imports of CommonJS and ES module JavaScript files, and Node.js core modules, do
+not need an import assertion; but they can be given a `type` of `'javascript'`
+if desired:
+
+```js
+const url = getUrlOfModuleToBeDynamicallyImported();
+
+let assertType = 'javascript';
+if (url.endsWith('.json')) {
+  assertType = 'json';
+} else if (url.endsWith('.wasm')) {
+  assertType = 'webassembly';
+}
```

Comment:
This doesn't take into account urls with a search query (`./file.json?cache=false`), a hash part (`./file.json#hash`), or `data:` urls. I think we should at least put a comment saying the implementation is not complete (or make the implementation complete).

Reply:
Good point, I've updated it to catch those cases.